pkg/virt-operator/resource/generate/components/secrets.go

```diff
@@ -295,10 +295,10 @@ func NextRotationDeadline(cert *tls.Certificate, ca *tls.Certificate, renewBefor
 }
 
 func ValidateSecret(secret *k8sv1.Secret) error {
-	if _, ok := secret.Data[bootstrap.CertBytesValue]; !ok {
+	if v, ok := secret.Data[bootstrap.CertBytesValue]; !ok || len(v) == 0 {
 		return fmt.Errorf("%s value not found in %s secret\n", bootstrap.CertBytesValue, secret.Name)
 	}
-	if _, ok := secret.Data[bootstrap.KeyBytesValue]; !ok {
+	if v, ok := secret.Data[bootstrap.KeyBytesValue]; !ok || len(v) == 0 {
```

Comment:
What I do not fully understand is that I would expect that `LoadCertificates` would fail with empty keys when it tries to parse the content. I wonder why that did not make the test pass either.

Reply:
That's what I'm wondering. It was a bit strange result. :-)